### PR TITLE
Grouped list fix

### PR DESCRIPTION
**Pull request by dzearing**

### CODE REVIEW

File: src/components/GroupedList/GroupHeader.tsx [hunk 1]
```diff
@@ -84,7 +74,7 @@ export class GroupHeader extends React.Component<IGroupHeader, IGroupHeaderState
             <button
               className='ms-GroupHeader-check'
               data-selection-toggle={ true }
-              onClick={ this._onToggleSelectGroup } >
```
Comment (yiminwu) [inline src/components/GroupedList/GroupHeader.tsx]: This name is confusing, should it just be _onSelectGroupClick?

File: src/components/GroupedList/GroupHeader.tsx [hunk 2]
```diff
@@ -84,7 +74,7 @@ export class GroupHeader extends React.Component<IGroupHeader, IGroupHeaderState
...
+              onClick={ this._onToggleSelectClick } >
```
Comment (yiminwu) [inline src/components/GroupedList/GroupHeader.tsx]: This name is confusing, should it just be _onSelectGroupClick?

Comment (dzearing) [inline src/components/GroupedList/GroupHeader.tsx]: it actually toggles selection for the group.

File: src/components/GroupedList/GroupHeader.tsx [hunk 3]
```diff
@@ -136,29 +123,23 @@ export class GroupHeader extends React.Component<IGroupHeader, IGroupHeaderState
...
-  private _onToggleSelectGroup(ev: React.MouseEvent) {
-    let { group, headerProps } = this.props;
-    let onToggleSelectGroup = headerProps && headerProps.onToggleSelectGroup;
+  private _onToggleSelectGroupClick(ev: React.MouseEvent) {
+    let { onToggleSelectGroup, group } = this.props;
 
-    if (onToggleSelectGroup) {
-      onToggleSelectGroup(group);
-    }
+    onToggleSelectGroup(group);
```
Comment (yiminwu) [inline src/components/GroupedList/GroupHeader.tsx]: need to have if

File: src/components/GroupedList/GroupedList.Props.ts [hunk 4]
```diff
@@ -181,15 +205,4 @@ export interface IGroupHeaderProps {
 
   /** Callback for when the group is selected. */
   onToggleSelectGroup?: (group: IGroup) => void;
-
-  /** Determines if the group selection check box is shown for collapsed groups. */
-  isCollapsedGroupSelectVisible?: boolean;
```
Comment (yiminwu) [inline src/components/GroupedList/GroupedList.Props.ts]: don't remove this.

File: src/components/GroupedList/GroupedListSection.tsx [hunk 5]
```diff
@@ -135,62 +139,48 @@ export class Group extends React.Component<IGroupProps, IGroupState> {
       headerProps,
       footerProps,
       viewport,
-      selectionMode
+      selectionMode,
+      onRenderGroupHeader = this._onRenderGroupHeader,
+      onRenderGroupFooter = this._onRenderGroupFooter
     } = this.props;
     let renderCount = group && getGroupItemLimit ? getGroupItemLimit(group) : Infinity;
-    let isFooterVisible = group && !group.children && !group.isCollapsed && !group.isShowingAll && group.count > renderCount;
```
Comment (yiminwu) [inline src/components/GroupedList/GroupedListSection.tsx]: take a look this.

File: src/components/GroupedList/GroupedListSection.tsx [hunk 6]
```diff
@@ -135,62 +139,48 @@ export class Group extends React.Component<IGroupProps, IGroupState> {
...
+      viewport: viewport,
+      selectionMode: selectionMode
+    };
+    let groupHeaderProps: IGroupDividerProps = assign({}, headerProps, dividerProps);
+    let groupFooterProps: IGroupDividerProps = assign({}, footerProps, dividerProps);
 
     return (
       <div
-        ref='root'
+        ref={ 'root' }
```
Comment (yiminwu) [inline src/components/GroupedList/GroupedListSection.tsx]: undo this.

File: src/demo/components/ExampleCard/ExampleCard.scss [hunk 7]
```diff
@@ -69,6 +69,7 @@
 .ExampleCard-example {
   max-height: 80vh;
   overflow: auto;
+  -webkit-overflow-scrolling: touch;
```
Comment (yiminwu) [inline src/demo/components/ExampleCard/ExampleCard.scss]: delete this.

**Pull request merged**